colossalai/fx/passes/concrete_info_prop.py

```diff
@@ -0,0 +1,305 @@
+from dataclasses import asdict
+from colossalai.fx.profiler import GraphInfo
+import torch
+import torch.fx
+from torch.fx.node import Node, Argument, Target
+from torch.utils._pytree import tree_map
+from typing import Any, List, Tuple, NamedTuple, Dict
+from torch.fx._compatibility import compatibility
+from colossalai.fx.profiler import profile_function, profile_module, profile_method, activation_size
+
+
+@compatibility(is_backward_compatible=True)
+class TensorMetadata(NamedTuple):
+    # TensorMetadata is a structure containing pertinent information
+    # about a tensor within a PyTorch program.
+
+    shape: torch.Size
+    dtype: torch.dtype
+    requires_grad: bool
+    stride: Tuple[int]
+    numel: int
+    is_tensor: bool
+    # TODO: we can add a list of sharding spec here, and record the sharding
+    # behaviour by appending sharding spec into list.
+
+
+def _extract_tensor_metadata(result: torch.Tensor) -> TensorMetadata:
+    """
+    Extract a TensorMetadata NamedTuple describing `result`.
+    """
+    shape = result.shape
+    dtype = result.dtype
+    requires_grad = result.requires_grad
+    stride = result.stride()
+    numel = result.numel()
+    is_tensor = True
+
+    return TensorMetadata(shape, dtype, requires_grad, stride, numel, is_tensor)
```

Comment:
You don't need these for concrete_info_prop

Reply:
okay